genai-perf/genai_perf/inputs/retrievers/synthetic_data_retriever.py

```diff
@@ -51,17 +51,30 @@ def retrieve_data(self) -> GenericDataset:
         files = self.config.synthetic_input_filenames or [DEFAULT_SYNTHETIC_FILENAME]
         synthetic_dataset = GenericDataset(files_data={})
 
+        use_system_prompts = self.config.num_system_prompts > 0
+        if use_system_prompts:
+            SyntheticPromptGenerator.create_system_prompts_pool(
+                self.config.tokenizer,
+                self.config.num_system_prompts,
+                self.config.system_prompt_length,
+            )
+
         for file in files:
             data_rows: List[DataRow] = []
 
             for _ in range(self.config.num_dataset_entries):
                 row = DataRow(texts=[], images=[])
-                prompt = SyntheticPromptGenerator.create_synthetic_prompt(
-                    self.config.tokenizer,
-                    self.config.prompt_tokens_mean,
-                    self.config.prompt_tokens_stddev,
-                )
                 for _ in range(self.config.batch_size_text):
+                    prompt = SyntheticPromptGenerator.create_synthetic_prompt(
+                        self.config.tokenizer,
+                        self.config.prompt_tokens_mean,
+                        self.config.prompt_tokens_stddev,
+                    )
+                    if use_system_prompts:
+                        system_prompt = (
+                            SyntheticPromptGenerator.get_random_system_prompt()
+                        )
+                        prompt = f"{system_prompt} {prompt}"
```

Comment:
Based on the requirements we were asked for, this is what's needed. We will likely work separately on different chat templates, e.g. in multi-turn chat.

In terms of this change though, it's just a prefix that gets prepended to each prompt. That allows benchmarking scenarios where schemes like prefix caching are used to evaluate the performance gain.

Reply:
That's a great callout. I asked Alec to weigh in. I called it the system prompt because that's what the requirement stated, so I assume that's the popular nomenclature. We definitely want to ensure clarity and avoid collisions.

Reply:
You're right. Alec stated: "In real life chat applications it is often the case the system prompt is the thing that will be shared between many different sessions as that is instructions to the agent that will stay constant."

He said that the implementation we have here is more of a shared prefix, which is fine for benchmarking. Ultimately, we may want to consider moving toward supporting it as a system prompt, but we don't really have that role-based templating at the moment, so this is a prefix now. I'll rename this to avoid confusion and naming collisions.